v2/cmd/controller/app/setup.go

```diff
@@ -253,6 +254,30 @@ func SetupControllerManager(ctx context.Context, setupLog logr.Logger, flgs Flag
 	return mgr
 }
 
+func getMetricsOpts(flags Flags) server.Options {
+	var metricsOptions server.Options
+	if flags.SecureMetrics {
+		metricsOptions = server.Options{
+			BindAddress:    flags.MetricsAddr,
+			SecureServing:  true,
+			FilterProvider: filters.WithAuthenticationAndAuthorization,
+			// Note that pprof endpoints are meant to be sensitive and shouldn't be exposed publicly.
+			ExtraHandlers: map[string]http.Handler{
```

Comment:
We probably should have a specific cmdline flag that enables this and it should default to off.

This is what [apiserver does](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/) via the `--profiling` argument, though it defaults it to true.

I'd argue we should have both `metrics.secure` and `metrics.profiling` flags in Helm, and two cmdline args in ASO to control these two bits.

Secure metrics shouldn't (IMO) _require_ that you expose pprof.